src/lib/circuit_value.test.ts

```diff
@@ -31,4 +40,28 @@ describe('circuit', () => {
       ])
     ).toThrow(/`mask` must have 0 or 1 true element, found 2/);
   });
+
+  it('can serialize Struct with array', async () => {
+    class MyStruct extends Struct({
+      values: Circuit.array(Field, 2),
+    }) {}
+
+    const original = new MyStruct({ values: [Field(0), Field(1)] });
+
+    const serialized = MyStruct.toJSON(original);
+    const reconstructed = MyStruct.fromJSON(serialized);
+
+    Circuit.assertEqual<MyStruct>(MyStruct, original, reconstructed);
+  });
+
+  it('can serialize nested Struct', async () => {
```

Comment:
should we maybe also test something like just to be sure? 

```ts
class OtherStruct extends Struct({
      x: Field,
}) {}

class MyStruct extends Struct({
      y: OtherStruct,
}) {}
```

Reply:
When I added this test, I got an identical stack trace using either a CircuitValue (in my case PrivateKey) or explicitly defined Struct, so I only posted the PrivateKey variant for brevity.

It makes sense to add an explicit nested Struct test too but if CircuitValue types will be migrated to Struct anyway it will likely become redundant.

Reply:
good catch!